app/src/main/java/com/android/shelfLife/ui/camera/CameraView.kt

```diff
@@ -101,39 +132,111 @@ fun BarcodeScannerScreen(
             selectedItem = Route.SCANNER)
       }) { paddingValues ->
         if (permissionGranted) {
-          // Display the camera preview
           Box(modifier = Modifier.fillMaxSize().padding(paddingValues)) {
-            // Remember the ROI rectangle
+            // State variables
+            val barcodeScanned = remember { mutableStateOf<String?>(null) }
+            val foodScanned = remember { mutableStateOf(false) }
+            val foodFacts = remember { mutableStateOf<FoodFacts?>(null) }
+            val isScanning by isScanningState
+            val searchInProgress = remember { mutableStateOf(false) }
+
+            // ROI calculation (same as before)
             val roiRectF = remember { mutableStateOf<RectF?>(null) }
+            val screenWidth = LocalContext.current.resources.displayMetrics.widthPixels
+            val screenHeight = LocalContext.current.resources.displayMetrics.heightPixels
+
+            val calculatedRoiRectF =
+                calculateRoiRectF(screenWidth.toFloat(), screenHeight.toFloat())
+            roiRectF.value = calculatedRoiRectF
 
             // Camera Preview
             CameraPreviewView(
                 modifier = Modifier.fillMaxSize(),
                 onBarcodeScanned = { scannedBarcode ->
-                  // Handle the scanned barcode
                   Log.d("BarcodeScanner", "Scanned barcode: $scannedBarcode")
                   beep()
-                  // Update ViewModel or navigate as needed
-                  foodFactsViewModel.searchByBarcode(scannedBarcode.toLong())
                   Toast.makeText(context, "Scanned barcode: $scannedBarcode", Toast.LENGTH_SHORT)
                       .show()
-                  // cameraViewModel.onBarcodeScanned(scannedBarcode)
-                },
-                onPreviewViewCreated = { previewView ->
-                  // Do nothing for now
+                  barcodeScanned.value = scannedBarcode
+                  isScanningState.value = false
+                  searchInProgress.value = true
                 },
-                roiRect = roiRectF.value ?: RectF(0f, 0f, 1f, 1f) // Default ROI
-                )
+                onPreviewViewCreated = { previewView -> },
```

Comment:
You could make this lambda cleaner
```suggestion
                onPreviewViewCreated = { _ },
```